### ISSUE TITLE
PostUploadHandler sometimes crashes during auto-upload of a draft

### ISSUE THREAD

**Issue by oguzkocer** [wordpress-mobile/WordPress-Android#10525]
Running the following steps sometimes end up with a crash for me:

1. Turn on airplane mode
2. Go to post list - drafts
3. Add a new draft with some title and/or content
4. Tap back which should say "Local draft"
5. Turn off airplane mode and immediately refresh the post list (not sure if this part is related or not)
6. Profit (crash)

Here is my test: https://cloudup.com/cRTp-q_JQfA

I was able to reproduce the crash a couple times, but it doesn't happen every time for me.

Edit: This test was run with a WordPress.com account.

```
--------- beginning of crash
2019-09-25 15:16:50.800 14479-14479/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wordpress.android, PID: 14479
    org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:527)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:509)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501)
        at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int org.wordpress.android.fluxc.model.PostModel.getLocalSiteId()' on a null object reference
        at org.wordpress.android.ui.uploads.PostUploadHandler.onPostChanged(PostUploadHandler.java:644)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501) 
        at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2019-09-25 15:16:50.800 14479-14845/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: pool-3-thread-20
    Process: org.wordpress.android, PID: 14479
    org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:527)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:509)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501)
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:527)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:509)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:451)
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:411)
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:384)
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:265)
        at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:41)
        at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32)
        at org.wordpress.android.fluxc.store.PostStore.handleRemoteAutoSavedPost(PostStore.java:1021)
        at org.wordpress.android.fluxc.store.PostStore.onAction(PostStore.java:632)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501) 
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.lang.IllegalStateException: post must not be null
        at org.wordpress.android.ui.posts.PostListEventListener.onPostChanged(PostListEventListener.kt:142)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:451) 
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:411) 
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:384) 
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:265) 
        at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:41) 
        at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32) 
        at org.wordpress.android.fluxc.store.PostStore.handleRemoteAutoSavedPost(PostStore.java:1021) 
        at org.wordpress.android.fluxc.store.PostStore.onAction(PostStore.java:632) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507) 
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501) 
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764)
```

**Comment by malinajirka** [comment]
I believe this crashes is caused by this https://github.com/wordpress-mobile/WordPress-Android/issues/8594 issue.

1. Invoke pushPost with PostA(localId = 1)
2. Invoke fetchList
3. Server processes step 1 -> PostA(remoteId = 1R) is Created on the server, but the client hasn't handled the response yet
4. Serve processes step 2 -> Server returns list which contains PostA(remoteId = 1R)
5. Client receives PostA(remoteId = 1R) but it doesn't know it's actually PostA(localId = 1) -> client creates PostA(remoteId = 1R, localId = **2**)
6. Client receives result of pushPost from step 1 -> PostA(remoteId = 1R, localId = 1) 
7. Client invokes remoteAutoSave on PostA(remoteId = 1R, localId = 1)
8. Client removes duplicate PostA(remoteId = 1R, localId = **1**) and keeps  PostA(remoteId = 1R, localId = **2**)
9. Client receives remoteAutoSave result but the post with localId 1 doesn't exist anymore => crash


I'll investigate this a bit more because I'm still not sure where do we perform step 8 -> who deletes the duplicated post. Ideally we'd fix the source of the issue instead of creating workarounds.

**Comment by oguzkocer** [comment]
Fixed in #10529.
